# FBB_Reproducibility GOATOOLS: A Python library for Gene Ontology analyses

Для воспроизведения результатов статьи "GOATOOLS: A Python library for Gene Ontology analyses" (https://www.nature.com/articles/s41598-018-28948-z) были использованы данные из Supplementary Table 1 в статье "Conserved epigenomic signals in mice and humans reveal immune basis of Alzheimer’s disease" (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4530583/).

Создавался файл с Ensembl ID, который в последствии подавался на вход различным тулам.

In [7]:
import pandas as pd
import numpy as np

df = pd.read_excel('nature14252-s1.xlsx', sheet_name='masterInfo.4.3.7.txt', header=1)
cols = [4,9]
new_df = df.drop(df.columns[cols], axis=1)
new_df['Ensembl ID'].to_csv('Ensembl_ID.txt', sep='\t', index=False)

Код ниже выводит количество генов в определённом кластере. 

In [6]:
tr_in = (new_df.Cluster.values == 'Transient Increase').sum()
con_in = (new_df.Cluster.values == 'Consistent Increase').sum()
late_in = (new_df.Cluster.values == 'Late Increase').sum()
tr_de = (new_df.Cluster.values == 'Transient Decrease').sum()
con_de = (new_df.Cluster.values == 'Consistent Decrease').sum()
late_de = (new_df.Cluster.values == 'Late Decrease').sum()
none = (new_df.Cluster.values == 'None').sum()
all_genes = tr_in + con_in + late_in + con_de + late_de + tr_de + none
print('Transient Increase = ' + str(tr_in))
print('Consistent Increase = ' + str(con_in))
print('Late Increase = ' + str(late_in))
print('Transient Decrease = ' + str(tr_de))
print('Consistent Decrease = ' + str(con_de))
print('Late Decrease = ' + str(late_de))
print('Total = ' + str(all_genes))

Transient Increase = 624
Consistent Increase = 794
Late Increase = 1397
Transient Decrease = 142
Consistent Decrease = 1029
Late Decrease = 1139
Total = 13836


К сожалению, из-за большого списка генов (13836) использовать API не рекомендовалось в DAVID 6.7 и 6.8, поэтому данные были поданы на вход вручную, которые впоследствии обрабатывались python.

In [8]:
df68 = pd.read_csv('6.8_DIRECT.txt', sep="\t")
df67 = pd.read_csv('6.7_ALL.txt', sep="\t")
df68['PValue'].to_csv('6.8_PValue.txt', sep='\t', index=False)
df67['PValue'].to_csv('6.7_PValue.txt', sep='\t', index=False)